test/dto/PartitionTest.cpp

```diff
@@ -0,0 +1,127 @@
+
+// Integration tests for partition operations
+// These assume three partitions with range partition schema: ["", c), [c, e), [e, "") 
+class PartitionTest {
+
+public:  // application lifespan
+    PartitionTest() : _client(k2::K23SIClientConfig()) { K2INFO("ctor");}
+    ~PartitionTest(){ K2INFO("dtor");}
+
+    // required for seastar::distributed interface
+    seastar::future<> gracefulStop() {
+        K2INFO("stop");
+        return std::move(_testFuture);
+    }
+
+    seastar::future<> start(){
+        K2INFO("start");
+
+        _testFuture = seastar::make_ready_future()
+        .then([this] () {
+            return _client.start();
+        })
+        .then([this] {
+            K2INFO("Creating test collection...");
+            std::vector<k2::String> rangeEnds;
+            rangeEnds.push_back(k2::dto::FieldToKeyString<k2::String>("c"));
+            rangeEnds.push_back(k2::dto::FieldToKeyString<k2::String>("e"));
+            rangeEnds.push_back("");
+
+            return _client.makeCollection(collname, std::move(rangeEnds));
+        })
+        .then([](auto&& status) {
+            K2EXPECT(status.is2xxOK(), true);
+        })
+        .then([this] { return runScenario01(); })
+        .then([this] {
+            K2INFO("======= All tests passed ========");
+            exitcode = 0;
+        })
+        .handle_exception([this](auto exc) {
+            try {
+                std::rethrow_exception(exc);
+            } catch (std::exception& e) {
+                K2ERROR("======= Test failed with exception [" << e.what() << "] ========");
+                exitcode = -1;
+            } catch (...) {
+                K2ERROR("Test failed with unknown exception");
+                exitcode = -1;
+            }
+        })
+        .finally([this] {
+            K2INFO("======= Test ended ========");
+            seastar::engine().exit(exitcode);
+        });
+
+        return seastar::make_ready_future();
+    }
+
+private:
+    int exitcode = -1;
+    seastar::future<> _testFuture = seastar::make_ready_future();
+
+    k2::K23SIClient _client;
+    k2::dto::PartitionGetter _pgetter;
+
+    
+public: // tests
+
+// get partition for key through range scheme
+seastar::future<> runScenario01() {
+    K2INFO("runScenario01");
+
+    return seastar::make_ready_future()
+    .then([this] {
+        K2INFO("get Partition for key with default reverse and exclusiveKey flag");
+        
+        k2::dto::Key key{.schemaName = "schema", .partitionKey = "a", .rangeKey = ""};
+        k2::dto::PartitionGetter::PartitionWithEndpoint& part = _pgetter.getPartitionForKey(key);
+        K2INFO("startkey:" << part.partition->startKey << ", endkey:" << part.partition->endKey << ", endpoint:" 
+               << *(part.partition->endpoints.begin()) << ", TXEndpoint:" << (*part.preferredEndpoint).getURL());
+        //K2EXPECT(part.partition->startKey, "");
+    });
+}
+
+};
+
+int main(int argc, char** argv) {
```

Comment:
I think you could have done this as a unit test instead of an integration test, but you don't need to change it now if you don't want to.